### PR TITLE
[58734] Extract work package info line into a Primer component and document it

### PR DESCRIPTION
**Pull request by bsatarnejad**
# Ticket
https://community.openproject.org/projects/openproject/work_packages/58734/activity

# What are you trying to accomplish?
Make a separate component that shows type, ID and status of a WP in one line. Therefore, this component can be used in different places like meeting agenda item, hover card, ...

## Screenshots
Before:
Screenshot 2024-11-20 at 12 34 37

After:
Screenshot 2024-11-20 at 12 25 58

# Merge checklist

- [ ] Added/updated tests
- [x] Added/updated documentation in Lookbook (patterns, previews, etc)
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

### CODE REVIEW

File: app/components/work_packages/info_line_component.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class WorkPackages::InfoLineComponent < ApplicationComponent
+  include OpPrimer::ComponentHelpers
+
+  def initialize(id:)
```
Comment (HDinger) [inline app/components/work_packages/info_line_component.rb]: I think it would be better to pass the WorkPackage directly.

**Review by HDinger — COMMENTED** [review]
Could you please rebase that on https://github.com/opf/openproject/pull/16989 and update the usages so that we see it in use?

**Comment by bsatarnejad** [comment]
> Could you please rebase that on #16989 and update the usages so that we see it in use?

Yes sure, I'll do that 👍

File: lookbook/docs/patterns/26-work-package-info-line.md.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- Work package relations tab
+- Work package header
+- ...
+
+### Code structure
+
+You can typically call the InfoLineComponent in views where work package details (Type, ID, and Status) need to be displayed in one line. You must pass a work package object containing valid data to it.
+
+```html
+    render(WorkPackages::InfoLineComponent.new(work_package: @work_package))
```
Comment (HDinger) [inline lookbook/docs/patterns/26-work-package-info-line.md.erb:33]: What I am actually missing as a user is to see how the component looks like. I don't know how much value it brings to explain how the component is called, as it is rather simple and very straight forward. In any case, please embedd a preview, like on [other patterns pages](https://github.com/opf/openproject/blob/dev/lookbook/docs/patterns/13-feedback-dialog-and-feedback-message.md.erb#L75).

File: app/components/work_packages/info_line_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  flex_layout do |flex|
+    flex.with_column(mr: 2) do
+      render(WorkPackages::HighlightedTypeComponent.new(work_package: @work_package, font_size: :small))
+    end
+    flex.with_column(mr: 2)  do
+      render(Primer::Beta::Text.new(font_size: :small, color: :muted)) { "##{@work_package.id}" }
+    end
+    flex.with_column do
+      render WorkPackages::StatusBadgeComponent.new(status: @work_package.status)
+    end
```
Comment (HDinger) [inline app/components/work_packages/info_line_component.html.erb]: Currently, the font-size is always set to `small` and the ID always `muted`. I guess, it makes sense to make at least the font_size configurable. The default could still be `small`.

File: lookbook/docs/patterns/26-work-package-info-line.md.erb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+## Anatomy
+
+The info line consists of three attributes that are displayed together:
+
+- Type
+- ID
+- Status
+
+Only the ID the only interactive element; it is clickable and points the work package. All other interaction has to come from the element using this work component.
```
Comment (HDinger) [inline lookbook/docs/patterns/26-work-package-info-line.md.erb:15]: Looking at that, the ID should be clickable, which is currently not the case.

**Review by HDinger — CHANGES_REQUESTED** [review]
Hi @bsatarnejad 
I have some smaller wishes for improvement for the component. Further, please also update the other occurences of the InfoLine that we are already aware of, in this PR (meetings, hovercard, ...)

File: lookbook/docs/patterns/26-work-package-info-line.md.erb (hunk 2 above)
Comment (bsatarnejad) [reply lookbook/docs/patterns/26-work-package-info-line.md.erb:33]: Hi @HDinger I added a preview in the [overview](https://github.com/opf/openproject/pull/17234/files#diff-c34c40a58d487841fe36a6a7c073d3aa157f3221a0056b45052804a97a3dbe1cR3) section.

Comment (HDinger) [reply lookbook/docs/patterns/26-work-package-info-line.md.erb:33]: Ah great, I missed that 👍

**Review by HDinger — APPROVED** [review]
Nice 👍 🌟

**Pull request merged**